### ISSUE TITLE
AbstractMethodError when debuggable is false

### ISSUE THREAD

**Issue by cf256** [muxinc/mux-stats-sdk-exoplayer#102]
Crash when running app with `debuggable false`.

Checkout the demo app linked here https://github.com/cf256/exoplayermux.
When debuggable is true for the debug flavor, everything is fine. When it is false the app crashes. 

Crash: 
` E  FATAL EXCEPTION: main
                         E  Process: no.rikstv.muxexoexample, PID: 14107
                         E  java.lang.AbstractMethodError: abstract method "void com.google.android.exoplayer2.analytics.AnalyticsListener.onMediaItemTransition(com.google.android.exoplayer2.analytics.AnalyticsListener$EventTime, com.go
                            ogle.android.exoplayer2.MediaItem, int)"
                         E      at com.google.android.exoplayer2.analytics.AnalyticsCollector.lambda$onMediaItemTransition$34(AnalyticsCollector.java:578)
                         E      at com.google.android.exoplayer2.analytics.-$$Lambda$AnalyticsCollector$DGKi9OUzTZtRXnRKtCVi5vrQrEQ.invoke(Unknown Source:8)
                         E      at com.google.android.exoplayer2.util.ListenerSet$ListenerHolder.invoke(ListenerSet.java:297)
                         E      at com.google.android.exoplayer2.util.ListenerSet.lambda$queueEvent$0(ListenerSet.java:184)
                         E      at com.google.android.exoplayer2.util.-$$Lambda$ListenerSet$NbKDn9xtItiyMgYZmjIx_Sv1FFQ.run(Unknown Source:6)
                         E      at com.google.android.exoplayer2.util.ListenerSet.flushEvents(ListenerSet.java:205)
                         E      at com.google.android.exoplayer2.util.ListenerSet.sendEvent(ListenerSet.java:220)
                         E      at com.google.android.exoplayer2.analytics.AnalyticsCollector.sendEvent(AnalyticsCollector.java:797)
                         E      at com.google.android.exoplayer2.analytics.AnalyticsCollector.onMediaItemTransition(AnalyticsCollector.java:575)
                         E      at com.google.android.exoplayer2.ExoPlayerImpl.lambda$updatePlaybackInfo$8(ExoPlayerImpl.java:1021)
                         E      at com.google.android.exoplayer2.-$$Lambda$ExoPlayerImpl$hCy9GuyuWqdycGpxOBepZjyxPeQ.invoke(Unknown Source:6)
                         E      at com.google.android.exoplayer2.util.ListenerSet$ListenerHolder.invoke(ListenerSet.java:297)
                         E      at com.google.android.exoplayer2.util.ListenerSet.lambda$queueEvent$0(ListenerSet.java:184)
                         E      at com.google.android.exoplayer2.util.-$$Lambda$ListenerSet$NbKDn9xtItiyMgYZmjIx_Sv1FFQ.run(Unknown Source:6)
                         E      at com.google.android.exoplayer2.util.ListenerSet.flushEvents(ListenerSet.java:205)
                         E      at com.google.android.exoplayer2.ExoPlayerImpl.updatePlaybackInfo(ExoPlayerImpl.java:1101)
                         E      at com.google.android.exoplayer2.ExoPlayerImpl.setMediaSourcesInternal(ExoPlayerImpl.java:1195)
                         E      at com.google.android.exoplayer2.ExoPlayerImpl.setMediaSources(ExoPlayerImpl.java:397)
                         E      at com.google.android.exoplayer2.ExoPlayerImpl.setMediaItems(ExoPlayerImpl.java:365)
                         E      at com.google.android.exoplayer2.BasePlayer.setMediaItems(BasePlayer.java:49)
                         E      at com.google.android.exoplayer2.BasePlayer.setMediaItem(BasePlayer.java:34)
                         E      at com.google.android.exoplayer2.SimpleExoPlayer.setMediaItem(SimpleExoPlayer.java:1322)
                         E      at no.rikstv.muxexoexample.MainActivity.initializePlayer(MainActivity.kt:82)
                         E      at no.rikstv.muxexoexample.MainActivity.onCreate(MainActivity.kt:35)
                         E      at android.app.Activity.performCreate(Activity.java:8000)
                         E      at android.app.Activity.performCreate(Activity.java:7984)
                         E      at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
                         E      at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3404)
                         E      at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3595)
                         E      at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
                         E      at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
                         E      at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
                         E      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
                         E      at android.os.Handler.dispatchMessage(Handler.java:106)
                         E      at android.os.Looper.loop(Looper.java:223)
                         E      at android.app.ActivityThread.main(ActivityThread.java:7660)
                         E      at java.lang.reflect.Method.invoke(Native Method)
                         E      at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
                         E      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)`

This is using Exoplayer 2.13.2 and mux 2.13.1. Reproduced on Pixel 3a running Android 11 (I'm pretty sure it is any device). It could be related to this issue (https://github.com/muxinc/mux-stats-sdk-exoplayer/issues/18?) maybe?

**Comment by tomkordic** [comment]
Hi @cf256 this is not an Mux SDK related issue, it is project configuration issue, running your sample project app I saw the crash on this line of code: https://github.com/cf256/exoplayermux/blob/main/app/src/main/java/no/rikstv/exoplayertest/MainActivity.kt#L79
The app crashes on actual ExoPlayer function, the reason it crashes is related to dexing artifact transform, issue explained here: https://github.com/google/ExoPlayer/issues/7349

If you add this line to your gradle.properties file: android.enableDexingArtifactTransform=false
the app will run just fine with debuggable set to false.

Let me know if this works for you.

**Comment by cf256** [comment]
Hi @tomkordic . Thanks for clearing that up, it did indeed fix my issue!
Thanks!

**Comment by denis-bezrukov** [comment]
@tomkordic We faced this issue as well and in our case `android.enableDexingArtifactTransform=false` didn't help - it fixed this particular crash, but app started crashing in other places. After investigation I found that the root cause is that we use only some parts of exoplayer, e.g. `exoplayer-core`,`exoplayer-ui`, `exoplayer-hls`. While mux sdk depends on `exoplayer` which depends on ton of other artifacts. Since mux sdk is distributed via github attachments as simple AAR file (without POM file), application is being built without some runtime dependencies. 

Currently I build muxsdk from sources with pom, but it would be much better if you will distribute artifacts with transitive dependencies (ideally via any maven repository or at least with uploading pom as github attachment too).

Thanks!
